hyperdome_server/web/share_mode.py

```diff
@@ -65,123 +66,114 @@ def __init__(self, common, web):
+                self.counselors_available[sid] += 1
+            elif other_user in self.counselors_available:
+                self.counselors_available[other_user] += 1
+            return 'Chat Ended'
+
+        @self.web.app.route("/counselor_signout", methods=["POST"])
+        def counselor_signout():
+            sid = request.form['user_id']
+            self.counselors_available.pop(sid, '')
             return "Success"
 
+        @self.web.app.route("/counselor_signin")
+        def counselor_signin():
+            # TODO authenticate
+            # user = load_user(request.form['username'])
+            sid = binascii.b2a_hex(os.urandom(15)).decode('utf-8')
+            # will use capacity variable for this later
+            self.counselors_available[sid] = 1
+            return sid
+
         @self.web.app.route("/generate_guest_id")
         def generate_guest_id():
-            return binascii.b2a_hex(os.urandom(15))
+            # TODO check for collisions
+            return binascii.b2a_hex(os.urandom(15)).decode('utf-8')
 
-        @self.web.app.route("/message_from_therapist", methods=['POST'])
-        @login_required
-        def message_from_therapist():
-            if current_user.username not in self.connected_guest:
-                return abort(401)
-            message = request.form['message']
-
-            guest_id = self.connected_guest[current_user.username]
-            self.pending_messages[guest_id] = (
-                self.pending_messages.get(
-                    guest_id, "") + message + "\n")
-            return "Success"
-
-        @self.web.app.route("/message_from_user", methods=['POST'])
+        @self.web.app.route("/send_message", methods=['POST'])
         def message_from_user():
             message = request.form['message']
-            guest_id = request.form['guest_id']
-            therapist_username = self.connected_therapist[guest_id]
-            self.pending_messages[therapist_username] = (
-                self.pending_messages.get(therapist_username, "")
-                + message + "\n")
+            user_id = request.form['user_id']
+            if user_id not in self.active_chat_user_map:
+                return "no chat", 404
+            other_user = self.active_chat_user_map[user_id]
+            if other_user in self.pending_messages:
+                self.pending_messages[other_user] += f"\n{message}"
+            elif other_user:  # may be empty string if other disconnected
+                self.pending_messages[other_user] = message
+            else:
+                return "user left", 404
             return "Success"
 
-        @self.web.app.route("/collect_guest_messages")
-        def collect_guest_messages():
-            guest_id = request.form['guest_id']
+        @self.web.app.route("/chat_status")
+        def chat_status():
+            user_id = request.form['user_id']
+            try:
+                if self.active_chat_user_map[user_id] == '':
```

Comment:
Ternary might cool here

Reply:
hey it's under the column limit for pep8

Fixed